concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limit/window/ImmutablePercentileSampleWindow.java

```diff
@@ -71,10 +72,15 @@ public long getCandidateRttNanos() {
 
     @Override
     public long getTrackedRttNanos() {
-        observedRtts.sort(Comparator.naturalOrder());
-        int rttIndex = (int) Math.round(observedRtts.size() * percentile);
+        long[] copyOfObservedRtts = new long[sampleCount];
```

Comment:
Do we need a sampleCount == 0 check here?

Reply:
Yes, we do. What value should we return if window contains no samples? Zero might hurt the limit, however this will happen only on servers that are extremely low in load

Reply:
Returning 0 is probably fine.  We filter out empty sample windows anyway so this actually won't be called.